crates/client/src/storage/fs.rs

```diff
@@ -76,7 +79,16 @@ impl FileSystemPackageStorage {
 }
 
 #[async_trait]
-impl PackageStorage for FileSystemPackageStorage {
+impl RegistryStorage for FileSystemPackageStorage {
+    async fn load_checkpoint(&self) -> Result<Option<SerdeEnvelope<MapCheckpoint>>> {
+        let contents = load(&self.base_dir.join("checkpoint")).await;
+        contents
+    }
+
+    async fn store_checkpoint(&self, checkpoint: SerdeEnvelope<MapCheckpoint>) -> Result<()> {
+        store(&self.base_dir.join("checkpoint"), checkpoint).await;
+        Ok(())
```

Comment:
This appears to be an unused result:

```suggestion
        store(&self.base_dir.join("checkpoint"), checkpoint).await
```